Mathlib/GroupTheory/Submonoid/Center.lean

```diff
@@ -50,9 +52,20 @@ theorem center_toSubsemigroup : (center M).toSubsemigroup = Subsemigroup.center
 
 variable {M}
 
+/-- The center of a multiplication with unit is commutative and associative. -/
+instance center.commMonoid : CommMonoid (center M) :=
+  { (center M).toMulOneClass, Subsemigroup.center.commSemigroup with }
```

Comment:
We have a bit of a problem here; this forms an instance diamond:
```lean
example : center.commMonoid.toMonoid = (center M).toMonoid := rfl -- fails due to npow not matching
```
Perhaps the thing to do is make this a def, and for now only have the int

Reply:
I think you've resolved this now? (See also [Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.236996.20Define.20the.20centre.20.2E.2E.2E.20for.20non-associative.20algebras) )